core/src/main/scala/caliban/schema/Schema.scala

```diff
@@ -73,9 +73,21 @@ trait Schema[-R, T] { self =>
   def resolve(value: T): Step[R]
 
   /**
-   * Defines if the type is considered optional or non-null. Should be false except for `Option`.
+   * Defines if the type can resolve to null or not.
+   * It is true if the type is nullable or if it can fail.
    */
-  def optional: Boolean = false
+  @deprecatedOverriding("this method will be made final. Override canFail and nullable instead", "2.6.1")
```

Comment:
It looks like we're going for a major release next. Change version to `2.7.0` in the message